lib/rubocop/cop/rspec/no_expectation_example.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks if an example includes any expectation.
+      #
+      # All RSpec's example and expectation methods are covered by default.
+      # If you are using your own custom methods,
+      # add the following configuration:
+      #
+      #   RSpec/NoExpectationExample:
+      #     ExampleMethodnames:
+      #       - custom_it
+      #     ExpectationMethodNames:
+      #       - custom_expect
+      #
+      # @example
+      #
+      #   # bad
+      #   it do
+      #     a?
+      #   end
+      #
+      #   # good
+      #   it do
+      #     expect(a?).to be_truthy
+      #   end
+      class NoExpectationExample < Base
+        extend AutoCorrector
+
+        include RangeHelp
+
+        DEFAULT_EXAMPLE_METHOD_NAMES = ::Set[
+          :example,
+          :fexample,
+          :fit,
+          :focus,
+          :fscenario,
+          :fspecify,
+          :it,
+          :its,
+          :pending,
+          :scenario,
+          :skip,
+          :specify,
+          :xexample,
+          :xit,
+          :xscenario,
+          :xspecify,
+        ]
+
+        DEFAULT_EXPECTATION_METHOD_NAMES = ::Set[
+          :expect_any_instance_of,
+          :expect,
+          :is_expected,
+          :should_not_receive,
+          :should_not,
+          :should_receive,
+          :should,
+        ]
+
+        MSG = 'No expectation found in this example.'
+
+        # @param [RuboCop::AST::Node] node
+        def on_send(node)
+          return unless example_method_names.include?(node.method_name)
+
+          block = argument_block_of(node)
+          return unless block
```

Comment:
it seems like this would be better implement in an `on_block` and then check if the send_note of that block is an example ?